syntax = "proto3";

package NAlice.NScenarios;

option go_package = "a.yandex-team.ru/alice/megamind/protos/scenarios";
option java_package = "ru.yandex.alice.megamind.protos.scenarios.begemot";
option java_outer_classname = "BegemotProto";

// NOTE(a-square): not all fields are reflected here, add them as needed

message TBegemotItemSelectorResult {
    message TGallery {
        message TItem {
            double Score = 1 [json_name = "score"];
            bool IsSelected = 2 [json_name = "is_selected"];
            string Alias = 3 [json_name = "alias"];
        }
        string GalleryName = 1 [json_name = "gallery_name"];
        repeated TItem Items = 2 [json_name = "items"];
    }
    repeated TGallery Galleries = 1 [json_name = "galleries"];
}

message TBegemotFixlistResult {
    message TValue {
        repeated string Intents = 1 [json_name = "Intents"];
    }

    message TMatch {
        string Key = 1 [json_name = "key"];
        TValue Value = 2 [json_name = "value"];
    }

    repeated TMatch Matches = 1 [json_name = "Matches"];
}
